WooCommerce/Classes/ViewRelated/Orders/Refund/RefundSubmissionUseCase.swift

```diff
@@ -275,20 +275,12 @@ private extension RefundSubmissionUseCase {
         // TODO: 5984 - tracks in-person refund error
         DDLogError("Failed to refund: \(error.localizedDescription)")
         // Informs about the error.
+        //TODO: Check which refund errors can't be retried, and use that to call nonRetryableError(from: self.rootViewController, error: cancelError)
```

Comment:
Great point on this TODO, do we have a way to tell if an error is non-retryable? In IPP, it just considers a failure case as non-retryable if the payment cancellation fails 🤔

Reply:
Android have this implemented already, I just didn't dig in to it enough yesterday to pull out their list. I was just going to steal that though! It's in the app, [Stripe have a bit of guidance](https://stripe.dev/stripe-terminal-ios/docs/Classes/SCPTerminal.html#/c:objc(cs)SCPTerminal(im)processRefund)...

> When processRefund fails, the SDK returns an error that either includes the failed [SCPRefund](https://stripe.dev/stripe-terminal-ios/docs/Classes/SCPRefund.html) or the [SCPRefundParameters](https://stripe.dev/stripe-terminal-ios/docs/Classes/SCPRefundParameters.html) that led to a failure. Your app should inspect the [SCPProcessRefundError](https://stripe.dev/stripe-terminal-ios/docs/Classes/SCPProcessRefundError.html) to decide how to proceed.
> 
> If the refund property is nil, the request to Stripe’s servers timed out and the refund’s status is unknown. We recommend that you retry processRefund with the original [SCPRefundParameters](https://stripe.dev/stripe-terminal-ios/docs/Classes/SCPRefundParameters.html).
> 
> If the [SCPProcessRefundError](https://stripe.dev/stripe-terminal-ios/docs/Classes/SCPProcessRefundError.html) has a failure_reason, the refund was declined. We recommend that you take action based on the decline code you received.

... but it's not exactly definite about which ones are retryable!